tests/foreman/virtwho/utils.py

```diff
@@ -0,0 +1,185 @@
+    running_stauts = ['is running', 'Active: active (running)']
+    stopped_status = ['is stopped', 'Active: inactive (dead)']
+    if ret == 0 and any(key in stdout for key in running_stauts):
+        return 'running'
+    elif ret == 0 and any(key in stdout for key in stopped_status):
+        return 'stopped'
+    else:
+        return 'undefined'
+
+
+def get_rhsm_logfile():
+    """Return a dictionary by analysing rhsm.log, and then can know
+        the virt-who configure file is deployed as expected or not.
+
+    The following keys will be included:
+    :[error]: how many error message exist, the expected value is 0.
+    :[reporter_id]: it can be defined by virtwho configure.
+    :[interval_time]: it can be defined by virtwho configure.
+    :[hypervisor_id]: it can be defined by virtwho configure.
+    :[hypervisor_name]: get to know the hypervisor's hostname.
+    :[guest_id]: get to know the guest_id.
+    """
+    ret, stdout = runcmd('cat /var/log/rhsm/rhsm.log')
+    data = dict()
+    data['error'] = len(re.findall(r'\[.*ERROR.*\]', stdout))
+    if "virtwho.main DEBUG" in stdout:
+        res = re.findall(r"reporter_id='(.*?)'", stdout)
+        if len(res) > 0:
+            data['reporter_id'] = res[0].strip()
+        res = re.findall(r"infinite loop with(.*?)seconds", stdout)
+        if len(res) > 0:
+            data['interval_time'] = int(res[0].strip())
+        res = re.findall(r'[^\s]*"hypervisorId": "(.*?)"', stdout)
+        if len(res) > 0:
+            data['hypervisor_id'] = res[0].strip()
+        res = re.findall(r'[^\s]*"name": "(.*?)"', stdout)
+        if len(res) > 0:
+            data['hypervisor_name'] = res[0].strip()
+        res = re.findall(r'[^\s]*guestId": "(.*?)"', stdout)
+        if len(res) > 0:
+            data['guest_id'] = res[0].strip()
+    return data
+
+
+def deploy_configure_by_command(command):
+    """Return the hypervisor_name and guest_name if deployed successfully.
+    :param str command: when created a configure file by UI, you can read
+        the deploy command line such as:
+        `hammer virt-who-config deploy --id 1 --organization-id 1`
+    :raises: VirtWhoError: If failed to deploy configure by this command.
+    """
+    virtwho_cleanup()
+    register_system(get_system('guest'))
+    ret, stdout = runcmd(command)
+    status = get_virtwho_status()
+    # Waiting for the h/g mapping report sent to satellite.
+    time.sleep(10)
+    data = get_rhsm_logfile()
+    if (status == 'running' and
+            data['error'] == 0 and 'hypervisor_id' in data):
+        hypervisor_name = data['hypervisor_id']
+        # Get the hostname for guest.
+        _, guest_name = runcmd(
+                'hostname',
+                system=get_system('guest'))
+        # Delete the hypervisor entry and always make sure it's new.
+        _, stdout = runcmd(
+            "hammer host list --search {}|sed '1,3d'|sed '$d'|cut -d '|' -f1"
+            .format(hypervisor_name))
```

Comment:
Can you use something from `robottelo.cli` here instead of calling hammer directly?